### PR TITLE
[bare-expo][ios] enable hermes

### PR DESCRIPTION
**Pull request by Kudo**
# Why

propose and demonstrate a way to configure hermes and works well with config-plugin. our previous discussion might be based on environment variables, but that is not ideal for config-plugin and eas-build. we need add custom environment variable support to eas-build to do that.

# How

this pr introduce a json file near Podfile and config-plugin could reliable update this json file.

# Test Plan

1. set `expo.jsEngine` as hermes or jsc in `podfile.properties.json`
2. `pod install` and `yarn ios`

### CODE REVIEW

File: apps/bare-expo/ios/Podfile [hunk 1]
```diff
@@ -10,6 +10,9 @@ require File.join(File.dirname(`node --print "require.resolve('react-native/pack
 require File.join(File.dirname(`node --print "require.resolve('@react-native-community/cli-platform-ios/package.json')"`), "native_modules")
 require File.join(File.dirname(`node --print "require.resolve('expo-modules-core/package.json')"`), "scripts/autolinking")
 
+require 'json'
+expo_config = JSON.parse(File.read('./expo_config.json')) rescue {}
```
Comment (EvanBacon) [inline apps/bare-expo/ios/Podfile:14]: Maybe something like `ios.config.json` or `podfile.config.json` so it follows the JS tooling name style. If we want this tied to expo, maybe `ios/.expo/<name>.config.json`

File: apps/bare-expo/ios/Podfile [hunk 2]
```diff
@@ -20,7 +23,7 @@ abstract_target 'BareExpoMain' do
   use_react_native!(
     :path => config[:reactNativePath],
     # to enable hermes on iOS, change `false` to `true` and then install pods
-    :hermes_enabled => false
+    :hermes_enabled => expo_config['jsEngine'] == 'hermes'
```
Comment (EvanBacon) [inline apps/bare-expo/ios/Podfile:26]: Maybe bind this to a namespace so the args can be passed directly?
```
{
  "react-native": {
    "hermes_enabled": true
  }
}
```

File: apps/bare-expo/ios/Podfile (hunk 1 above)
Comment (EvanBacon) [reply apps/bare-expo/ios/Podfile:14]: Maybe we should also manage this file using a mod so other plugins can safely make changes to it

Comment (Kudo) [reply apps/bare-expo/ios/Podfile:14]: or `podfile.props.json` like `gradle.properties`. (idea from @brentvatne). preferring `podfile.<something>.json` personally and not necessary tied to expo.

File: apps/bare-expo/ios/Podfile (hunk 2 above)
Comment (Kudo) [reply apps/bare-expo/ios/Podfile:26]: just want to align the naming we did in android gradle.properties and app.json. does it make sense?

File: apps/bare-expo/ios/Podfile (hunk 1 above)
Comment (brentvatne) [reply apps/bare-expo/ios/Podfile:14]: podfile.properties.json maybe?

Comment (Kudo) [reply apps/bare-expo/ios/Podfile:14]: > podfile.properties.json maybe?

this name looks good to me. going to use this name and if there are any comments, could revise if needed.

**Review by brentvatne — APPROVED** [review]

**Pull request merged**